shared_model/backend/protobuf/query_responses/proto_role_permissions_response.hpp

```diff
@@ -40,24 +42,32 @@ namespace shared_model {
       RolePermissionsResponse(RolePermissionsResponse &&o)
           : RolePermissionsResponse(std::move(o.proto_)) {}
 
-      const PermissionNameCollectionType &rolePermissions() const override {
+      const interface::RolePermissionSet &rolePermissions() const override {
         return *rolePermissions_;
       }
 
+      std::string toString() const override {
+        return detail::PrettyStringBuilder()
+            .init("RolePermissionsResponse")
+            .appendAll(permissions::toString(rolePermissions()),
+                       [](auto p) { return p; })
+            .finalize();
+      }
+
      private:
       template <typename T>
       using Lazy = detail::LazyInitializer<T>;
 
       const iroha::protocol::RolePermissionsResponse &rolePermissionsResponse_{
           proto_->role_permissions_response()};
 
-      const Lazy<PermissionNameCollectionType> rolePermissions_{[this] {
+      const Lazy<interface::RolePermissionSet> rolePermissions_{[this] {
         return boost::accumulate(
             rolePermissionsResponse_.permissions(),
-            PermissionNameCollectionType{},
+            interface::RolePermissionSet{},
             [](auto &&permissions, const auto &permission) {
-              permissions.emplace_back(permission);
-              return std::move(permissions);
+              permissions.set(interface::permissions::fromOldR(permission));
+              return std::forward<decltype(permissions)>(permissions);
```

Comment:
Is forward here really necessary? Why not keeping move?

Reply:
Just similarly to that line:
https://github.com/hyperledger/iroha/blob/632f14e0a305e7cde3f14683a6f7d9660e6c36c4/shared_model/backend/protobuf/commands/impl/proto_create_role.cpp#L25